custom_metrics/sass.js

```diff
@@ -0,0 +1,295 @@
+		//         we are not taking that into account here. This applies to functions too (below)
+		ret.mixins[name] = {
+			// Default values can't have commas, so split should work
+			args: args?.split(/\s*,\s*/), // TODO separate name from default value?
+			calls: scss.match(RegExp("@include\\s+" + name + "\\b", "gi"))?.length
+		};
+	});
+
+	// Custom functions
+	ret.functions = {};
+	let lastName; // used to track down the nearest @return after a @function
+	scss.replace(/@function\s+([\w-]+)(?:\((.+?)\))?|@return (.+)/g, ($0, name, args, returnValue) => {
+		if ($0.indexOf("@return") === 0 && lastName) {
+			ret.functions[lastName].returns = ret.functions[lastName].returns || [];
+			ret.functions[lastName].returns.push(returnValue);
+			return $0;
+		}
+
+		lastName = name;
+		ret.functions[name] = {
+			// Default values can't have commas, so split should work
+			args: args?.split(/\s*,\s*/), // TODO separate name from default value?
+			calls: scss.match(RegExp("\\b" + name + "\\(", "gi"))?.length
+		};
+	});
+
+	// TODO Measure usage of Sass functions
+	ret.functionCalls = {};
+	let functionCallRegex = new RegExp(`(?<![-.\\w])(${SassFunctions.join("|").replace(".", "\\.")})\\(`, "gi");
+	scss.replace(functionCallRegex, ($0, name) => {
+		ret.functionCalls[name] = (ret.functionCalls[name] || 0) + 1;
+	});
+	ret.functionCalls = sortObject(ret.functionCalls);
+
+	// Conditionals
+	ret.ifs = [];
+	scss.replace(/@(?:(else )?if\s+?(.+?)|@else)(?=\s*\{)/g, ($0, elseIf, test) => {
+		if ($0 === "@else") {
+			// Else for previous condition
+			let obj = ret.ifs.pop();
+			console.log(ret.ifs, obj);
+			obj.hasElse = true;
+			ret.ifs.push(obj);
+		}
+		else if (elseIf) {
+			// Else if for previous condition
+			let obj = ret.ifs.pop();
+			obj.elseIfs = obj.elseIfs || [];
+			obj.elseIfs.push(test);
+			ret.ifs.push(obj);
+		}
+		else {
+			// New conditional
+			ret.ifs.push({test});
+		}
+	});
+
+	// @each, @for, @while Loops
+	ret.eaches = {};
+	scss.replace(/@each\s+(.+ in .+)(?=\s*\{)/g, ($0, args) => {
+		ret.eaches[args] = (ret.eaches[args] || 0) + 1;
+	});
+
+	ret.fors = {};
+	// console.log("@for", scss);
+	scss.replace(/@for\s+(.+)(?=\s*\{)/g, ($0, args) => {
+		ret.fors[args] = (ret.fors[args] || 0) + 1;
+	});
+
+	ret.whiles = {};
+	scss.replace(/@while\s+(.+)(?=\s*\{)/g, ($0, args) => {
+		ret.whiles[args] = (ret.whiles[args] || 0) + 1;
+	});
+
+	// @extend
+	ret.extends = {};
+	scss.replace(/@extend\s*(.+)(?=\s*(?:[;}]|$))/g, ($0, selector) => {
+		ret.extends[selector] = (ret.extends[selector] || 0) + 1;
+	});
+
+	// @error
+	ret.errors = scss.match(/@error (.+)/g)?.map(e => e.slice(7));
+
+	// CSS variables that are set with Sass variables
+	// Note that this will fail on multiline values (it will return the first line only)
+	ret.variablesCombined = {value: {}, name: {}};
+	scss.replace(/(?<=^|\s)--([\w-]+):\s*(.*#\{.*\$.+\}.*)\s*$/gm, ($0, name, value) => {
```

Comment:
```suggestion
	scss.replace(/(?<=^|\s)--([\w-]+):\s*(.*#\{.+\}.*)\s*$/gm, ($0, name, value) => {
```

Suggest not requiring the `$` to trigger this, just capture any time `#{}` is used for Sass-defined custom props.

Reply:
Done!